File: lng_modeling/half_active/nlmpc_config__stateFcn.m
function dxdt = nlmpc_config__stateFcn(x,u,p)
    %
    % x: (1) Longitudinal Position
    %    (2) Body Vertical Displacement
    %    (3) Front Wheel Vertical Displacement
    %    (4) Rear Wheel Vertical Displacement
    %    (5) Body Pitch Angle
    %    (6) Front Wheel Angle
    %    (7) Rear Wheel Angle
    %    (8) Velocity
    %    (9) Body Vertical Velocity
    %    (10) Front Wheel Vertical Velocity
    %    (11) Rear Wheel Vertical Velocity
    %    (12) Body Pitch Angular Velocity
    %    (13) Front Wheel Angular Velocity
    %    (14) Rear Wheel Angular Velocity
    %
    % u: (1) front torque
    %    (2) rear torque
    %    (3) front sus
    %    (4) rear sus
    %
    % d: (1) longitudinal position of front wheel center
    %    (2) longitudinal position of rear wheel center
    %    (3) vertical position of front wheel center
    %    (4) vertical position of rear wheel center
    %    (5) gradient of longitudinal position of front wheel center
    %    (6) gradient of longitudinal position of rear wheel center
    %    (7) gradient of vertical position of front wheel center
    %    (8) gradient of vertical position of rear wheel center
    %
    
    % Copyright 2023 The MathWorks, Inc.
    %% Vehicle parameter
    dt = 0.01;     % control period
    m_b = 960;     % [kg]      body mass
    m_wf = 40;       % [kg]      wheel mass
    m_wr = m_wf;       % [kg]      wheel mass
    k_longf = 592180*2;
    k_longr = 394790*2;
    k_sf = 30000;   % [N/m]     front spring stiffness
    k_sr = 25000;   % [N/m]     rear spring stiffness
    k_wf = 270000;   % [N/m]     tire stiffness
    k_wr = k_wf;   % [N/m]     tire stiffness
    c_longf = 2360*2;
    c_longr = 2117*2;
    c_sf = 7000;    % [N/(m/s)] front damping
    c_sr = 6000;    % [N/(m/s)] rear damping
    c_wf = 1000;       % [N/(m/s)] tire damping
    c_wr = c_wf;       % [N/(m/s)] tire damping
    wb = 2.45;     % wheel base
    L_f = wb * 23/55;      % [m]       front length
    L_r = wb * 32/55;      % [m]       rear length
    r = 0.55/2;           % [m]   radius of wheel
    I_b = m_b*(wb/2)^2;      % [kgm^2]   body inertia moment
    I_wf = (m_wf*r^2)/2;     % [kgm^2]   wheel inertia moment
    I_wr = (m_wr*r^2)/2;     % [kgm^2]   wheel inertia moment

    g = 9.80665;
    pHorizon = 5;

    persistent A B E disc_func
    if isempty(A)
        A = [                       0,                          0,                   0,                   0,                              0, 0, 0,                        1,                          0,                   0,                   0,                              0, 0, 0;
                                    0,                          0,                   0,                   0,                              0, 0, 0,                        0,                          1,                   0,                   0,                              0, 0, 0;
                                    0,                          0,                   0,                   0,                              0, 0, 0,                        0,                          0,                   1,                   0,                              0, 0, 0;
                                    0,                          0,                   0,                   0,                              0, 0, 0,                        0,                          0,                   0,                   1,                              0, 0, 0;
                                    0,                          0,                   0,                   0,                              0, 0, 0,                        0,                          0,                   0,                   0,                              1, 0, 0;
                                    0,                          0,                   0,                   0,                              0, 0, 0,                        0,                          0,                   0,                   0,                              0, 1, 0;
                                    0,                          0,                   0,                   0,                              0, 0, 0,                        0,                          0,                   0,                   0,                              0, 0, 1;
             -(k_longf + k_longr)/m_b,                          0,                   0,                   0,                              0, 0, 0, -(c_longf + c_longr)/m_b,                          0,                   0,                   0,                              0, 0, 0;
                                    0,         -(k_sf + k_sr)/m_b,            k_sf/m_b,            k_sr/m_b,     -(L_f*k_sf - L_r*k_sr)/m_b, 0, 0,                        0,         -(c_sf + c_sr)/m_b,            c_sf/m_b,            c_sr/m_b,     -(L_f*c_sf - L_r*c_sr)/m_b, 0, 0;
                                    0,                  k_sf/m_wf, -(k_sf + k_wf)/m_wf,                   0,                (L_f*k_sf)/m_wf, 0, 0,                        0,                  c_sf/m_wf, -(c_sf + c_wf)/m_wf,                   0,                (L_f*c_sf)/m_wf, 0, 0;
                                    0,                  k_sr/m_wr,                   0, -(k_sr + k_wr)/m_wr,               -(L_r*k_sr)/m_wr, 0, 0,                        0,                  c_sr/m_wr,                   0, -(c_sr + c_wr)/m_wr,               -(L_r*c_sr)/m_wr, 0, 0;
                                    0, -(L_f*k_sf - L_r*k_sr)/I_b,      (L_f*k_sf)/I_b,     -(L_r*k_sr)/I_b, -(k_sf*L_f^2 + k_sr*L_r^2)/I_b, 0, 0,                        0, -(L_f*c_sf - L_r*c_sr)/I_b,      (L_f*c_sf)/I_b,     -(L_r*c_sr)/I_b, -(c_sf*L_f^2 + c_sr*L_r^2)/I_b, 0, 0;
                                    0,                          0,                   0,                   0,                              0, 0, 0,                        0,                          0,                   0,                   0,                              0, 0, 0;
                                    0,                          0,                   0,                   0,                              0, 0, 0,                        0,                          0,                   0,                   0,                              0, 0, 0];
        
        B = [     0,      0,       0,        0;
                  0,      0,       0,        0;
                  0,      0,       0,        0;
                  0,      0,       0,        0;
                  0,      0,       0,        0;
                  0,      0,       0,        0;
                  0,      0,       0,        0;
                  0,      0,       0,        0;
                  0,      0,   1/m_b,    1/m_b;
                  0,      0, -1/m_wf,        0;
                  0,      0,       0,  -1/m_wr;
                  0,      0, L_f/I_b, -L_r/I_b;
             1/I_wf,      0,       0,        0;
                  0, 1/I_wr,       0,        0];
        
        E = [          0,           0,         0,         0,           0,           0,         0,         0;
                       0,           0,         0,         0,           0,           0,         0,         0;
                       0,           0,         0,         0,           0,           0,         0,         0;
                       0,           0,         0,         0,           0,           0,         0,         0;
                       0,           0,         0,         0,           0,           0,         0,         0;
                       0,           0,         0,         0,           0,           0,         0,         0;
                       0,           0,         0,         0,           0,           0,         0,         0;
             k_longf/m_b, k_longr/m_b,         0,         0, c_longf/m_b, c_longr/m_b,         0,         0;
                       0,           0,         0,         0,           0,           0,         0,         0;
                       0,           0, k_wf/m_wf,         0,           0,           0, c_wf/m_wf,         0;
                       0,           0,         0, k_wr/m_wr,           0,           0,         0, c_wr/m_wr;
                       0,           0,         0,         0,           0,           0,         0,         0;
                       0,           0,         0,         0,           0,           0,         0,         0;
                       0,           0,         0,         0,           0,           0,         0,         0];
        
        % discretization
        disc_func = @(tau,Mat) (-pinv(A)*expm(A.*(dt-tau)))*Mat;

        % A = expm(A.*dt);
        % B = disc_func(dt,B) - disc_func(0,B);
        % E = disc_func(dt,E) - disc_func(0,E);
    end
    

    %% set parameters through horizon
    persistent param_flag last_param last_d first_states first_d current_d current_dis current_mileage_f current_mileage_r current_wheel_traj_f current_wheel_traj_r delta_x
    
    current_d = p(1:8);
    if isempty(last_param)
        param_flag = p(end);  % number of iteration
        last_param = p;
        current_d = p(1:8);  % current_disturbance
        current_dis = p(9:8+pHorizon+10);   % wheel trajectory constraints
        current_mileage_f = p(9+pHorizon+10:8+(pHorizon+10)*2);   % wheel trajectory constraints
        current_mileage_r = p(9+(pHorizon+10)*2:8+(pHorizon+10)*3);   % wheel trajectory constraints
        current_wheel_traj_f = [p(9+(pHorizon+10)*3:8+(pHorizon+10)*4)];   % wheel trajectory constraints
        current_wheel_traj_r = [p(9+(pHorizon+10)*4:8+(pHorizon+10)*5)];   % wheel trajectory constraints
        first_states = x;
        first_d = current_d;
        last_d = current_d;
    elseif any(param_flag ~= p(end))
        param_flag = p(end);
        last_param = p;
        current_d = p(1:8);
        current_dis = p(9:8+pHorizon+10);
        current_mileage_f = p(9+pHorizon+10:8+(pHorizon+10)*2);
        current_mileage_r = p(9+(pHorizon+10)*2:8+(pHorizon+10)*3);
        current_wheel_traj_f = [p(9+(pHorizon+10)*3:8+(pHorizon+10)*4)];
        current_wheel_traj_r = [p(9+(pHorizon+10)*4:8+(pHorizon+10)*5)];
        first_states = x;
        first_d = current_d;
        last_d = current_d;
    else
        check_x = round(x*1e10);
        check_first_state = round(first_states*1e10);
        if any(check_x ~= check_first_state)
            delta_x = x - first_states;
            front_wheel_rotation = delta_x(6)*r;
            rear_wheel_rotation  = delta_x(7)*r;
            
            current_d(1) = first_d(1) + makima(current_mileage_f,current_dis,front_wheel_rotation);  % x_disf
            current_d(2) = first_d(2) + makima(current_mileage_r,current_dis,rear_wheel_rotation);  % x_disr
            current_d(3) = round(makima(current_dis,current_wheel_traj_f,current_d(1)-first_d(1))*1e05)*1e-05;
            current_d(4) = round(makima(current_dis,current_wheel_traj_r,current_d(2)-first_d(2))*1e05)*1e-05;

            duration_length_f = dt*x(13)*r/2;
            temp_duration_f = [(current_d(1)-first_d(1))-duration_length_f,(current_d(1)-first_d(1)),(current_d(1)-first_d(1))+duration_length_f];
            temp_displacment_f = round(makima(current_dis,current_wheel_traj_f,temp_duration_f)*1e05)*1e-05;
            temp_gradient_f = [sum(diff(temp_duration_f))/2;sum(diff(temp_displacment_f))/2];
            duration_length_r = dt*x(14)*r/2;
            temp_duration_r = [(current_d(2)-first_d(2))-duration_length_r,(current_d(2)-first_d(2)),(current_d(2)-first_d(2))+duration_length_r];
            temp_displacment_r = round(makima(current_dis,current_wheel_traj_r,temp_duration_r)*1e05)*1e-05;
            temp_gradient_r = [sum(diff(temp_duration_r))/2;sum(diff(temp_displacment_r))/2];

            current_d(5) = (x(13)*r)*(temp_gradient_f(1)^2/sum(temp_gradient_f.^2));
            current_d(6) = (x(14)*r)*(temp_gradient_r(1)^2/sum(temp_gradient_r.^2));
            current_d(7) = (x(13)*r)*(temp_gradient_f(2)^2/sum(temp_gradient_f.^2));
            current_d(8) = (x(14)*r)*(temp_gradient_r(2)^2/sum(temp_gradient_r.^2));
            % current_d(5)
            % current_d(6)
            % current_d(7)
            % current_d(8)
        end
    end
    
    G = [                                     0;
                                              0;
                                              0;
                                              0;
                                              0;
                                              0;
                                              0;
                                              0;
                                             -1;
                                             -1;
                                             -1;
                                              0;
        -sin(atan(current_d(7)/current_d(5)))/r;
        -sin(atan(current_d(8)/current_d(6)))/r];

    % avoid nan
    if any(last_d(1,1) ~= current_d(1,1))
    else
        G(13) = 0;
    end
    if any(last_d(2,1) ~= current_d(2,1))
    else
        G(14) = 0;
    end

    last_d = current_d;
    
    dxdt = A*x + B*u + E*current_d + G*g;

end

File: lng_modeling/half_active/codegen/mex/nlmpcmoveCodeGeneration/nlmpc_config__stateFcn.c
/*
 * Academic License - for use in teaching, academic research, and meeting
 * course requirements at degree granting institutions only.  Not for
 * government, commercial, or other organizational use.
 *
 * nlmpc_config__stateFcn.c
 *
 * Code generation for function 'nlmpc_config__stateFcn'
 *
 */

/* Include files */
#include "nlmpc_config__stateFcn.h"
#include "makima.h"
#include "nlmpcmoveCodeGeneration_data.h"
#include "nlmpcmoveCodeGeneration_emxutil.h"
#include "nlmpcmoveCodeGeneration_types.h"
#include "rt_nonfinite.h"
#include "sumMatrixIncludeNaN.h"
#include "mwmathutil.h"
#include <emmintrin.h>
#include <string.h>

/* Variable Definitions */
static boolean_T A_not_empty;

static boolean_T last_param_not_empty;

/* Function Definitions */
void nlmpc_config__stateFcn(const real_T x[14], const real_T u[4],
                            const real_T p[309], real_T dxdt[14])
{
  static real_T A[196];
  static real_T E[112];
  static real_T b_current_dis[60];
  static real_T b_current_mileage_f[60];
  static real_T b_current_mileage_r[60];
  static real_T b_current_wheel_traj_f[60];
  static real_T b_current_wheel_traj_r[60];
  static real_T B[56];
  static real_T b_first_states[14];
  static real_T b_first_d[8];
  static real_T last_d[8];
  static real_T b_param_flag;
  emxArray_boolean_T *b_x;
  real_T check_x[14];
  real_T delta_x[14];
  real_T b_current_d[8];
  real_T b_y[3];
  real_T c_y[2];
  real_T temp_gradient_f[2];
  real_T temp_gradient_r[2];
  real_T duration_length_f;
  real_T temp_duration_f_tmp;
  int32_T i;
  int32_T k;
  boolean_T *x_data;
  emlrtHeapReferenceStackEnterFcnR2012b(emlrtRootTLSGlobal);
  /*  In a 2D environment with standard XY axis, the vehicle is a circular disc
   */
  /*  (20 meters in diamater).  Two thrusts are to the left and right of the */
  /*  center.  Tilting (theta) is defined as positive to left and negative to */
  /*  the right (0 means robot is vertical). */
  /*  */
  /*  x: (1) x position of the center of gravity in m */
  /*     (2) y position of the center of gravity in m */
  /*     (3) theta (tilt with respect to the center of gravity) in radian */
  /*     (4) dxdt */
  /*     (5) dydt */
  /*     (6) dthetadt */
  /*  */
  /*  u: (1) thrust on the left, in Newton */
  /*     (2) thrust on the right, in Newton */
  /*  */
  /*  The continuous-time model is valid only if the vehicle above or at the */
  /*  ground (y>=10). */
  /*  Copyright 2023 The MathWorks, Inc. */
  /*     %% Vehicle parameter */
  /*  control period */
  /*  [kg]      body mass */
  /*  [kg]      wheel mass */
  /*  [N/m]     front spring stiffness */
  /*  [N/m]     rear spring stiffness */
  /*  [N/m]     tire stiffness */
  /*  [N/(m/s)] front damping */
  /*  [N/(m/s)] rear damping */
  /*  [N/(m/s)] tire damping */
  /*  wheel base */
  /*  [m]       front length */
  /*  [m]       rear length */
  /*  [m]   radius of wheel */
  /*  [kgm^2]   body inertia moment */
  /*  [kgm^2]   wheel inertia moment */
  if (!A_not_empty) {
    memcpy(&A[0], &dv[0], 196U * sizeof(real_T));
    A_not_empty = true;
    memcpy(&B[0], &dv1[0], 56U * sizeof(real_T));
    memcpy(&E[0], &dv2[0], 112U * sizeof(real_T));
    /*  discretization */
    /*  A = expm(A.*dt); */
    /*  B = disc_func(dt,B) - disc_func(0,B); */
    /*  E = disc_func(dt,E) - disc_func(0,E); */
  }
  /*     %% set parameters through horizon */
  memcpy(&b_current_d[0], &p[0], 8U * sizeof(real_T));
  if (!last_param_not_empty) {
    b_param_flag = p[308];
    last_param_not_empty = true;
    memcpy(&b_current_d[0], &p[0], 8U * sizeof(real_T));
    memcpy(&b_current_dis[0], &p[8], 60U * sizeof(real_T));
    memcpy(&b_current_mileage_f[0], &p[68], 60U * sizeof(real_T));
    memcpy(&b_current_mileage_r[0], &p[128], 60U * sizeof(real_T));
    memcpy(&b_current_wheel_traj_f[0], &p[188], 60U * sizeof(real_T));
    memcpy(&b_current_wheel_traj_r[0], &p[248], 60U * sizeof(real_T));
    memcpy(&b_first_states[0], &x[0], 14U * sizeof(real_T));
    memcpy(&b_first_d[0], &b_current_d[0], 8U * sizeof(real_T));
    memcpy(&last_d[0], &b_current_d[0], 8U * sizeof(real_T));
  } else if (b_param_flag != p[308]) {
    b_param_flag = p[308];
    memcpy(&b_current_d[0], &p[0], 8U * sizeof(real_T));
    memcpy(&b_current_dis[0], &p[8], 60U * sizeof(real_T));
    memcpy(&b_current_mileage_f[0], &p[68], 60U * sizeof(real_T));
    memcpy(&b_current_mileage_r[0], &p[128], 60U * sizeof(real_T));
    memcpy(&b_current_wheel_traj_f[0], &p[188], 60U * sizeof(real_T));
    memcpy(&b_current_wheel_traj_r[0], &p[248], 60U * sizeof(real_T));
    memcpy(&b_first_states[0], &x[0], 14U * sizeof(real_T));
    memcpy(&b_first_d[0], &b_current_d[0], 8U * sizeof(real_T));
    memcpy(&last_d[0], &b_current_d[0], 8U * sizeof(real_T));
  } else {
    boolean_T exitg1;
    boolean_T y;
    emxInit_boolean_T(&b_x);
    k = b_x->size[0];
    b_x->size[0] = 14;
    emxEnsureCapacity_boolean_T(b_x, k);
    x_data = b_x->data;
    for (k = 0; k < 14; k++) {
      duration_length_f = muDoubleScalarRound(x[k] * 1.0E+10);
      x_data[k] = (duration_length_f != duration_length_f);
    }
    y = false;
    k = 0;
    exitg1 = false;
    while ((!exitg1) && (k <= 13)) {
      if (x_data[k]) {
        y = true;
        exitg1 = true;
      } else {
        k++;
      }
    }
    emxFree_boolean_T(&b_x);
    if (y) {
      __m128d r;
      for (k = 0; k <= 12; k += 2) {
        r = _mm_loadu_pd(&b_first_states[k]);
        _mm_storeu_pd(&delta_x[k], _mm_sub_pd(_mm_loadu_pd(&x[k]), r));
      }
      __m128d r1;
      real_T temp_duration_f[3];
      real_T d_y[2];
      b_current_d[0] = b_first_d[0] + makima(b_current_mileage_f, b_current_dis,
                                             delta_x[5] * 0.275);
      /*  x_disf */
      b_current_d[1] = b_first_d[1] + makima(b_current_mileage_r, b_current_dis,
                                             delta_x[6] * 0.275);
      /*  x_disr */
      b_current_d[2] =
          muDoubleScalarRound(makima(b_current_dis, b_current_wheel_traj_f,
                                     b_current_d[0] - b_first_d[0]) *
                              100000.0) *
          1.0E-5;
      b_current_d[3] =
          muDoubleScalarRound(makima(b_current_dis, b_current_wheel_traj_r,
                                     b_current_d[1] - b_first_d[1]) *
                              100000.0) *
          1.0E-5;
      duration_length_f = 0.001 * x[12] * 0.275 / 2.0;
      temp_duration_f_tmp = b_current_d[0] - b_first_d[0];
      temp_duration_f[0] = temp_duration_f_tmp - duration_length_f;
      temp_duration_f[1] = temp_duration_f_tmp;
      temp_duration_f[2] = temp_duration_f_tmp + duration_length_f;
      b_makima(b_current_dis, b_current_wheel_traj_f, temp_duration_f, b_y);
      b_y[0] = muDoubleScalarRound(b_y[0] * 100000.0);
      b_y[1] = muDoubleScalarRound(b_y[1] * 100000.0);
      b_y[2] = muDoubleScalarRound(b_y[2] * 100000.0);
      c_y[0] = temp_duration_f_tmp - temp_duration_f[0];
      c_y[1] = temp_duration_f[2] - temp_duration_f_tmp;
      r = _mm_loadu_pd(&b_y[0]);
      r1 = _mm_set1_pd(1.0E-5);
      _mm_storeu_pd(&b_y[0], _mm_mul_pd(r, r1));
      b_y[2] *= 1.0E-5;
      d_y[0] = b_y[1] - b_y[0];
      d_y[1] = b_y[2] - b_y[1];
      temp_gradient_f[0] = sumColumnB(c_y) / 2.0;
      temp_gradient_f[1] = sumColumnB(d_y) / 2.0;
      duration_length_f = 0.001 * x[13] * 0.275 / 2.0;
      temp_duration_f_tmp = b_current_d[1] - b_first_d[1];
      temp_duration_f[0] = temp_duration_f_tmp - duration_length_f;
      temp_duration_f[1] = temp_duration_f_tmp;
      temp_duration_f[2] = temp_duration_f_tmp + duration_length_f;
      b_makima(b_current_dis, b_current_wheel_traj_r, temp_duration_f, b_y);
      b_y[0] = muDoubleScalarRound(b_y[0] * 100000.0);
      b_y[1] = muDoubleScalarRound(b_y[1] * 100000.0);
      b_y[2] = muDoubleScalarRound(b_y[2] * 100000.0);
      c_y[0] = temp_duration_f_tmp - temp_duration_f[0];
      c_y[1] = temp_duration_f[2] - temp_duration_f_tmp;
      r = _mm_loadu_pd(&b_y[0]);
      _mm_storeu_pd(&b_y[0], _mm_mul_pd(r, r1));
      b_y[2] *= 1.0E-5;
      d_y[0] = b_y[1] - b_y[0];
      d_y[1] = b_y[2] - b_y[1];
      temp_gradient_r[0] = sumColumnB(c_y) / 2.0;
      temp_gradient_r[1] = sumColumnB(d_y) / 2.0;
      r = _mm_loadu_pd(&temp_gradient_f[0]);
      _mm_storeu_pd(&c_y[0], _mm_mul_pd(r, r));
      duration_length_f = sumColumnB(c_y);
      b_current_d[4] =
          x[12] * 0.275 *
          (temp_gradient_f[0] * temp_gradient_f[0] / duration_length_f);
      r = _mm_loadu_pd(&temp_gradient_r[0]);
      _mm_storeu_pd(&c_y[0], _mm_mul_pd(r, r));
      temp_duration_f_tmp = sumColumnB(c_y);
      b_current_d[5] =
          x[13] * 0.275 *
          (temp_gradient_r[0] * temp_gradient_r[0] / temp_duration_f_tmp);
      b_current_d[6] =
          x[12] * 0.275 *
          (temp_gradient_f[1] * temp_gradient_f[1] / duration_length_f);
      b_current_d[7] =
          x[13] * 0.275 *
          (temp_gradient_r[1] * temp_gradient_r[1] / temp_duration_f_tmp);
      /*  current_d(5) */
      /*  current_d(6) */
      /*  current_d(7) */
      /*  current_d(8) */
    }
  }
  memset(&check_x[0], 0, 8U * sizeof(real_T));
  check_x[8] = -1.0;
  check_x[9] = -1.0;
  check_x[10] = -1.0;
  check_x[11] = 0.0;
  check_x[12] =
      -muDoubleScalarSin(muDoubleScalarAtan(b_current_d[6] / b_current_d[4])) /
      0.275;
  check_x[13] =
      -muDoubleScalarSin(muDoubleScalarAtan(b_current_d[7] / b_current_d[5])) /
      0.275;
  if (!(last_d[0] != b_current_d[0])) {
    check_x[12] = 0.0;
  }
  if (!(last_d[1] != b_current_d[1])) {
    check_x[13] = 0.0;
  }
  memcpy(&last_d[0], &b_current_d[0], 8U * sizeof(real_T));
  /*  G(isnan(G)) = 0; */
  /*  G = disc_func(dt,G) - disc_func(0,G); */
  for (k = 0; k < 14; k++) {
    duration_length_f = 0.0;
    for (i = 0; i < 14; i++) {
      duration_length_f += A[k + 14 * i] * x[i];
    }
    temp_duration_f_tmp = 0.0;
    for (i = 0; i < 8; i++) {
      temp_duration_f_tmp += E[k + 14 * i] * b_current_d[i];
    }
    dxdt[k] = ((duration_length_f +
                (((B[k] * u[0] + B[k + 14] * u[1]) + B[k + 28] * u[2]) +
                 B[k + 42] * u[3])) +
               temp_duration_f_tmp) +
              check_x[k] * 9.80665;
  }
  /*  dxdt = x + dxdt*dt; */
  /*  if sum(isnan(current_d)) ~= 0 */
  /*      disp(current_d) */
  /*  end */
  /*  dxdt(isnan(dxdt)) = 0; */
  emlrtHeapReferenceStackLeaveFcnR2012b(emlrtRootTLSGlobal);
}

void nlmpc_config__stateFcn_init(void)
{
  last_param_not_empty = false;
  A_not_empty = false;
}

/* End of code generation (nlmpc_config__stateFcn.c) */
